examples/ResNet-Example.ipynb

```diff
@@ -0,0 +1,200 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "fbd304b1",
+   "metadata": {},
+   "source": [
+    "# Using Transformers from Huggingface\n",
+    "This is an example notebook of how to use Huggingface models with ZnNL"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b42c9519",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# import os\n",
+    "# os.environ['CUDA_VISIBLE_DEVICES'] = '-1'\n",
+    "\n",
+    "import znnl as nl\n",
+    "\n",
+    "import numpy as np\n",
+    "import optax\n",
+    "\n",
+    "from znnl.models import HuggingFaceFlaxModel\n",
+    "from transformers import ResNetConfig, FlaxResNetForImageClassification\n",
+    "import jax\n",
+    "print(jax.default_backend())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dba15f7c",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "data_generator = nl.data.CIFAR10Generator(2)\n",
+    "\n",
+    "# Input data needs to have shape (num_points, channels, height, width)\n",
```

Comment:
This is quite a funny problem. I also encountered it in some of our other projects. Is it worth setting this to the default? Just a question not something for this PR.

Reply:
Do you mean the data generator of 2 points, or the swapped axes of the input?

Reply:
The way they pass images with the channels in the first non-batch axis

Reply:
I found this very strange actually, as you still have to define the shape similar to as we currently do it. But the actual data is passed differently. 
We could theoretically adapt to that, but we could also implement e.g. a kwarg in the data generators to transform data to HF input shapes.
What do you think?

Reply:
For me, our approach makes more sense because for each pixel we assign N numbers for colour. This approach would say that each colour get M pixels which I find funny. In some sense it is up to the user to know what they want their data to look like, but we do provide generators for this purpose. I would prefer to stick to a single shape for those generators though. I think the correct thing to do would be to stick to the Hugging face structure as this is the unchangeable thing. Our models thusfar are all defined at initialization so swapping an axis is easy.